sdk/lib/atl/atlwin.h

```diff
@@ -575,6 +577,18 @@ class CWindow
         return ::GetDlgItemText(m_hWnd, nID, lpStr, nMaxCount);
     }
 
+    UINT GetDlgItemText(int nID, CSimpleString& string)
+    {
+        HWND item = GetDlgItem(nID);
+        int len = ::GetWindowTextLength(item);
+        if (len > 0)
+        {
+            GetDlgItemText(nID, string.GetBuffer(len+1), len+1);
+            string.ReleaseBuffer();
```

Comment:
If you catch the length from GetDlgItemText, you could pass that to ReleaseBuffer